### ISSUE TITLE
AADL models are not syntaxically correct, please fix them !

### ISSUE THREAD

**Issue by AtirChaudary** [pok-kernel/pok#4]
this happens when I run Makefile 
there is a syntax error in your code can you help me on this ?

**Comment by juli1** [comment]
Having a trace/comment might help :)

**Comment by AtirChaudary** [comment]
atir@ubuntu:~/Desktop/pok/examples/arinc653-events$ ls
Makefile  model.aadl  model.aadl~  receive.c
atir@ubuntu:~/Desktop/pok/examples/arinc653-events$ make
/home/atir/Desktop/POK/pok-master/misc/pok-toolchain.pl --arinc653 --models=model.aadl --no-run
AADL models are not syntaxically correct, please fix them !
make: *** [build] Error 1
atir@ubuntu:~/Desktop/pok/examples/arinc653-events$

**Comment by juli1** [comment]
looks like a problem with @yoogx and ocarina!

**Comment by AtirChaudary** [comment]
how can buffer and blackboard can run then ?

they both are running but events queueing are not :(
i have reinstalled the ocarina but still this issue is there

**Comment by AtirChaudary** [comment]
there is not much help on the internet.You are the last person who can solve this :)

**Comment by juli1** [comment]
I appreciate the compliment. Please note that POK and Ocarina are open source and contributions are more than appreciated.

**Comment by AtirChaudary** [comment]
i have corrected them thanks for your time

**Comment by yoogx** [comment]
@AtirChaudary, can you please send us a patch with your corrections ? These would also help us providing an updated version of the tool. Thanks

**Comment by kuanhsunchen** [comment]
@AtirChaudary @yoogx, is this issue solved to the end? I also have the same error. It would be nice if there is a patch.

**Comment by yoogx** [comment]
No, I did not hear back from @AtirChaudary since last August. And I lack resources to update these models.

**Comment by AtirChaudary** [comment]
see the .aadl file and see which functions you are using in your model.There are some mistakes like where se define language if = C but it should be (C) because thats how functions are accepting it. so i use it for my own and modified it but i am not working on this anymore i moved to other project 
hope this helps @yoogx  @kuanhsunchen

**Comment by gchamp20** [comment]
It seems most of the errors were introduced when this [commit ](https://github.com/OpenAADL/ocarina/commit/1427953c3d0a828578e775f91f63bd899fd193d5#diff-2c49c9d214e7b262243383d15069b909) was merge into ocarina. Mainly, it seems Module_Major_Frame, Partition_Slots, Slots_Allocation from ARINC653:: has to be replaced by POK::Major_Frame, POK::Slots, POK::Slots_Allocation. Plus what @AtirChaudary said about source_language => (C).

However, after correcting the model i'm getting the error from the script:  

ARINC653 deployment file is NOT conform to XSD schema

These are the format errors:

**Comment by juli1** [comment]
Can you please send:
1. The version of ocarina you are using
2. The version of pok you are using
3. The commands you are using

Having a way to reproduce the bug will definitively help us to fix it :-)

**Comment by gchamp20** [comment]
Sorry for the late reply.

1 - Ocarina 2017.1 (Working Copy from rfb77e27) --> [Github binaries](https://github.com/OpenAADL/ocarina/releases)
2 - Latest pok master (commit ddca1c4)
3 - `make` in examples/arinc653-events or any other arinc653-* test.

The build scripts hide the error output from the subcommand, so I've executed 
`ocarina  -k no-assert -k arinc653 -aadlv2 -f -g pok_c model.aadl $POK_PATH/misc/aadl-library.aadl ` to debug the AADL model and `../../../misc/check-arinc653-conffile.pl cpu_arinc653-conf.xml` to validate the XML configuration files after I got the model compiling.

**Comment by juli1** [comment]
I will look into this over the week end.

**Comment by juli1** [comment]
follow-up on this.

The reason is because the property set changed. Obviously, the ocarina compiler would need some love to be fixed and generate the HM policy based on the new property set. It will have to read the literals from the new property set and so on.

I managed to generate `arinc653-blackboard` but the generated code needs to be manually modified to compile (adding `POK_NEEDS_THREADS` and `POK_NEEDS_FUNCTION_TIME_GET` in the `deployment.h` file).

This is a long change that requires some time, please submit a patch to fix the issue and/or try to get commercial support if this is something you really need.

**Comment by wenting83** [comment]
> * based on the new properties.

Yes, exactly. Any update? Thank you!
